nni/typehint.py

```diff
@@ -18,6 +18,8 @@
 else:
     from typing_extensions import Literal, TypedDict
 
+from typing_extensions import TypeGuard
```

Comment:
In "required.txt": `typing_extensions >= 4.0.0 ; python_version < "3.8"`
Should we bump it to python 3.10?

Reply:
I think typing-extensions is already in the dependency tree. We can just make it required regardless of python versions.

Reply:
Seems both pytorch and tensorflow depends on typing extensions. Then I'm OK.

Reply:
I'm not familiar with typing_extensions. Is there advantage to use like `typing.Literal` over `typing_extensions.Literal`?
If the answer is no, seems we should stop using `nni.typehint.Literal` and use `typing_extensions.Literal` instead.